creator-node/src/routes/tracks.js

```diff
@@ -58,139 +60,208 @@ module.exports = function (app) {
     return successResponse({ 'track_segments': trackSegments })
   }))
 
-  /** given track metadata object, create track and share track metadata with network
-    * - return on success: temporary ID of track
-    * - return on failure: error if linked segments have not already been created via POST /track_content
-    */
-  app.post('/tracks', authMiddleware, nodeSyncMiddleware, handleResponse(async (req, res) => {
+  /**
+   * Given track metadata object, upload and share metadata to IPFS. Return metadata multihash if successful.
+   * Error if associated track segments have not already been created and stored.
+   */
+  app.post('/tracks/metadata', authMiddleware, preMiddleware, nodeSyncMiddleware, handleResponse(async (req, res) => {
     // TODO - input validation
-    const metadataJSON = req.body
-
-    if (!metadataJSON.owner_id) {
-      return errorResponseBadRequest('Metadata blob must include owner_id')
-    }
+    const metadataJSON = req.body.metadata
 
-    // get file multihashes from metadata blob; confirm each has an associated file or error
-    if (!metadataJSON.track_segments) {
-      return errorResponseBadRequest("Metadata blob must include list of multihashes as 'segments'")
+    if (!metadataJSON.owner_id || !metadataJSON.track_segments || !Array.isArray(metadataJSON.track_segments)) {
+      return errorResponseBadRequest('Metadata object must include owner_id and track_segments array')
     }
 
-    // get single file per multihash or error if DNE
-    let segmentFiles = []
-    for await (const segment of metadataJSON.track_segments) {
-      // TODO[SS] error check
-      //  - check if properties exist, if right format, if valid multihash, if valid duration
-      const segmentMultihash = segment.multihash
-
-      const file = await models.File.findOne({
-        where: {
-          multihash: segmentMultihash,
-          cnodeUserUUID: req.userId
-        }
-      })
+    // Ensure each segment multihash in metadata obj has an associated file, else error.
+    await Promise.all(metadataJSON.track_segments.map(async segment => {
```

Comment:
@SidSethi i think the Promise.all is redundant. this will block on each segment so putting it inside the await. The alternative is to await on the Promise.all() and then iterate through files once it resolves and make sure there are no nulls